src/main/java/core/basesyntax/service/impl/OperationStrategyImpl.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.OperationStrategy;
+import core.basesyntax.service.operation.OperationHandler;
+import core.basesyntax.service.operation.impl.OperationBalanceHandler;
+import core.basesyntax.service.operation.impl.OperationPurchaseHandler;
+import core.basesyntax.service.operation.impl.OperationReturnHandler;
+import core.basesyntax.service.operation.impl.OperationSupplyHandler;
+import java.util.HashMap;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<FruitTransaction.Operation, OperationHandler> operationHandlerMap;
+
+    public OperationStrategyImpl() {
+        this.operationHandlerMap = new HashMap<>();
```

Comment:
It's better to use dependency injection to provide instances of OperationHandler implementations instead of directly initializing them in the constructor.